course_classification/helpers.py

```diff
@@ -3,45 +3,77 @@
+    date = getattr(course, key, None)
+    if date:
+        return abs((date - today).days)
+    else:
+        return float('inf')
 
+    
 def get_course_ctgs(courses):
     """
-        Return dict with categories and its courses
+    Return dict with categories and its courses, sorted by enrollment and date criteria.
     """
     ret_courses = OrderedDict()
-    ret_courses['featured'] = []
     mc_courses= {}
+    today = timezone.now()  # Get timezone-aware current datetime
+    featured_courses = []
     if CourseClassification.objects.filter(is_featured_course=True).exists():
         for course in courses:
             try:
                 course_class = CourseClassification.objects.get(course_id=course.id)
+                course_start = getattr(course, 'start', None)
                 if course_class.is_featured_course:
-                    ret_courses['featured'].append(course)
+                    featured_courses.append(course)
                     if course_class.MainClass:
-                        mc_courses[course.id] = {'name': course_class.MainClass.name, 'logo': course_class.MainClass.logo}
+                        days_left = (course_start - today).days
+                        mc_courses[course.id] = {'name': course_class.MainClass.name,'logo': course_class.MainClass.logo,'days_left': days_left}
+                else:
+                    if course_class.MainClass:
+                        days_left = (course_start - today).days
+                        mc_courses[course.id] = {'name': course_class.MainClass.name, 'logo': course_class.MainClass.logo, 'days_left': days_left}
             except CourseClassification.DoesNotExist:
                 pass
-        if ret_courses['featured']:
-            return ret_courses, mc_courses
-    del ret_courses['featured']
+    if featured_courses:
+        # Apply classification and sorting to featured courses
+        ret_courses['featured'] = classify_and_sort_courses(featured_courses, today)
+        # Early return to match original behavior
+        return ret_courses, mc_courses
 
-    for main in CourseCategory.objects.all().order_by('sequence'):
-        ret_courses[main.id] = {'id':main.id,'name':main.name, 'seq':main.sequence,'show_opt':main.show_opt, 'courses':[]}
-    mc_courses= {}
-    for course in courses:
-        try:
-            course_class = CourseClassification.objects.get(course_id=course.id)
-            if course_class.MainClass:
-                mc_courses[course.id] = {'name': course_class.MainClass.name, 'logo': course_class.MainClass.logo}
-            for ctg in course_class.course_category.all():
-                ret_courses[ctg.id]['courses'].append(course)
-        except CourseClassification.DoesNotExist:
-            pass
+    # If no featured courses, proceed to build categories
+    categories = CourseCategory.objects.all().order_by('sequence')
+    if categories.exists():
+        for main in categories:
```

Comment:
Si se puede renombrar esta variable iterable en el `for`, sería de ayuda para entender el código. Como está ahora toda categoría es una "main", lo cual es raro. Podría ser `cat` o `ctg`.